File: rx-codelab/src/main/java/com/jraska/rx/codelab/RxLogging.java
package com.jraska.rx.codelab;

import io.reactivex.plugins.RxJavaPlugins;

public class RxLogging {

  public static void enableObservableSubscribeLogging() {
    RxJavaPlugins.setOnObservableSubscribe((observable, observer) -> {
      System.out.println(observable.getClass().getName() + ".onSubscribe(" + observer.getClass().getSimpleName() + ")");
      return observer;
    });
  }
}

File: rx-codelab/src/main/java/com/jraska/rx/codelab/RxLogging.kt
package com.jraska.rx.codelab

import io.reactivex.plugins.RxJavaPlugins

object RxLogging {

  fun enableObservableSubscribeLogging() {
    RxJavaPlugins.setOnObservableSubscribe { observable, observer ->
      println(observable.javaClass.getName() + ".onSubscribe(" + observer.javaClass.getSimpleName() + ")")
      return@setOnObservableSubscribe observer
    }
  }
}
